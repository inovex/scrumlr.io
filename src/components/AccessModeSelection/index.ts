import AccessModeSelection from './AccessModeSelection';
export default AccessModeSelection;
export * from './AccessModeSelection';
