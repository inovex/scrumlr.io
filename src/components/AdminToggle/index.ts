export * from './AdminToggle';

import AdminToggle from './AdminToggle';
export default AdminToggle;
