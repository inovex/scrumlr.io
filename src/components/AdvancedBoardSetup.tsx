import React from 'react';
import Typography from '@material-ui/core/Typography';
import FormGroup from '@material-ui/core/FormGroup';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import { useTranslation } from 'react-i18next';

export interface AdvancedBoardSetupProps {
  admissionControl: boolean;
  encryptedData: boolean;
  onSetAdmissionControl: (enabled: boolean) => void;
  onSetEncryptedData: (enabled: boolean) => void;
}

export const AdvancedBoardSetup: React.FC<AdvancedBoardSetupProps> = ({ admissionControl, encryptedData, onSetAdmissionControl, onSetEncryptedData }) => {
  const { t } = useTranslation('advancedBoardSetup');

  return (
    <>
      <ExpansionPanel>
        <ExpansionPanelSummary>
          <Typography>{t('heading')}</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <div>
            <Typography>{t('helper')}</Typography>
            <FormGroup>
              <FormControlLabel
                onClick={(event) => event.stopPropagation()}
                onFocus={(event) => event.stopPropagation()}
                control={
                  <Switch
                    checked={admissionControl}
                    onChange={(event) => {
                      onSetAdmissionControl(event.target.checked);
                    }}
                  />
                }
                label={t('admissionControl.heading')}
              />
              <Typography color="textSecondary">{t('admissionControl.helper')}</Typography>

              <FormControlLabel
                onClick={(event) => event.stopPropagation()}
                onFocus={(event) => event.stopPropagation()}
                control={<Switch checked={encryptedData} onChange={(event) => onSetEncryptedData(event.target.checked)} disabled={!admissionControl} />}
                label={t('encryptData.heading')}
              />
              <Typography color="textSecondary">{t('encryptData.helper')}</Typography>
            </FormGroup>
          </div>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </>
  );
};

export default AdvancedBoardSetup;
