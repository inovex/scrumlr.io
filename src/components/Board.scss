@import "common";

.board {
  --n: 1;

  display:grid;
  grid-auto-flow: column;
  grid-auto-columns: calc(100%/min(var(--real-columns), var(--n)));
  overflow-x:auto;
  scroll-snap-type: x mandatory;

  height: 100vh;
}

@for $i from 1 through 12 {
  @media screen and (min-width: $i*$min-column-width) {
    .board {
      --n: #{$i};
    }
  }
}