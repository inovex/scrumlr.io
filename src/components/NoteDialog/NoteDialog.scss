@import "constants/style";
@import "components/Note/Note.scss";

.note-dialog__header {
  font-size: 48px;
  color: $color-white;
  width: max-content;
  border-bottom: solid 3px var(--accent-color);
  padding-bottom: $padding--default;
  margin: $margin--default 0 $margin--large 0;
  letter-spacing: $letter-spacing--large;
  line-height: $line-height--large;
}

.note-dialog__note {
  padding: $padding--default $padding--medium;
  position: relative;
  border-radius: $note__border-radius;
  background-color: $color-white;
  box-shadow: 0 6px 9px 0 $note__box-shadow-color;
  transition: transform 0.3s ease-in-out, box-shadow 0.3s ease-in-out;
  list-style-type: none;
  z-index: $note-dialog-z-index;
  min-width: calc(240px + 10vw);
  width: calc(100vw - 36px);
  max-width: 380px;
}
@media (prefers-color-scheme: dark) {
  .note-dialog__note {
    background-color: $color-dark-mode-note;
    box-shadow: 0 6px 9px 0 $note__box-shadow-color-dark-mode;
  }
}
@media #{$menu-desktop} {
  .note-dialog__note {
    max-width: 440px;
    width: calc(100vw - 32px);
  }
}

.note-dialog__note--own-card::before {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  width: $note__indicator-width;
  height: $note__indicator-height;
  border-radius: 0 $note__indicator-width $note__indicator-width 0;
  background: var(--accent-color);
}

.note-dialog__content {
  display: flex;
  width: 100%;
}

.note-dialog__text {
  background: none;
  margin: 0;
  min-height: 3 * $line-height--medium;
  width: 100%;
  color: $color-black;
  font-size: $text-size;
  letter-spacing: $letter-spacing--medium;
  line-height: $line-height--medium;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  margin-bottom: $note__text-margin-bottom;
}
.note-dialog__text:focus {
  outline: 1px solid $color-dark-gray;
  outline-offset: 6px;
}
@media (prefers-color-scheme: dark) {
  .note-dialog__text {
    color: $color-white;
  }
  .note-dialog__text:focus {
    outline: 1px solid $color-white;
  }
}

.note-dialog__author {
  display: flex;
  align-items: center;
  // reset css from figure-element:
  margin: 0;
  padding: 0;
  border: 0;
  font: inherit;
  vertical-align: baseline;
}

.note-dialog__author-image {
  height: $note__author-image-size;
  width: $note__author-image-size;
  border-radius: 100%;
}

.note-dialog__author-name {
  margin-left: $note__author-margin-left;
  color: $color-dark-gray;
  font-size: $note__author-name-size;
  letter-spacing: $letter-spacing--small;
  line-height: $line-height--small;
}

//Options
.note-dialog__options {
  list-style-type: none;
  position: absolute;
  margin: 0;
  bottom: 10px;
  right: 25px;
}

@media #{$menu-desktop} {
  .note-dialog__options {
    top: 0;
    right: -80px - $margin--default;
  }
}

.note-dialog__option {
  box-sizing: border-box;
  display: block;
}

.note-dialog__option--not-editable {
  display: none;
}

.note-dialog__option + .note-dialog__option {
  margin-top: $margin--small;
}

.note-dialog__option-button {
  height: 32px;
}

.note-dialog__options-icon {
  display: flex;
  justify-content: center;
  width: 32px;
  height: 32px;
}

.note-dialog__options-icon svg {
  fill: #fff;
  width: 20px;
  height: 20px;
  align-self: center;
}
