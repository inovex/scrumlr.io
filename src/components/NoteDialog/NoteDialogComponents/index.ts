import {NoteDialogNote as Note} from "./NoteDialogNote";
import {NoteDialogHeader as Header} from "./NoteDialogHeader";

export const NoteDialogComponents = {
  Header,
  Note,
};
