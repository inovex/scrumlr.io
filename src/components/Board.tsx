import React from 'react';
import './Board.scss';
import {getColorClassName} from "../constants/colors";

const Board = ({ children }: any) => {
    const colors = React.Children.map(children, (child) => child.props.color);

    return (
        <>
            <style>
                {`.board { --board__columns: ${React.Children.count(children)} }`}
            </style>
            <div className="board">
                <div className={`board__spacer-left ${getColorClassName(colors[0])}`} />
                    {children}
                <div className={`board__spacer-right ${getColorClassName(colors[colors.length - 1])}`} />
            </div>
        </>
    )
};

export default Board;