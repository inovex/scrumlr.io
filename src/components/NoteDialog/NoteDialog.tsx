import "./NoteDialog.scss";
import avatar from "assets/avatar.png";
import Portal from "components/Portal/Portal";
import classNames from "classnames";
import Parse from "parse";
import {NoteProps} from "components/Note/Note";
import IconButton from "components/IconButton/IconButton";
import {ReactComponent as deleteIcon} from "assets/icon-delete.svg";
import React from "react";
import {Color, getColorClassName} from "constants/colors";

export interface NoteDialogProps extends NoteProps {
  show: boolean;
  editable?: boolean;
  authorName: string;
  onClose: () => void;
  onDelete: () => void;
  onEdit: (text: string) => void;
  columnName: string;
  columnColor: string;
}

const NoteDialog = ({show, text, authorId, editable, authorName, columnName, columnColor, onClose, onDelete, onEdit}: NoteDialogProps) => {
  if (!show) {
    return null;
  }

  const [noteText, setNoteText] = React.useState(text);

  const handleChangeNotetext = (e: React.FocusEvent<HTMLElement>) => {
    const textContent = e.target.textContent as string;
    if (editable) {
      onEdit(textContent);
      setNoteText(textContent);
    }
  };

  const handleDelete = () => {
    if (editable) {
      onDelete();
      onClose();
    }
  };

  return (
    <Portal onClose={onClose}>
      <div className={`note-dialog ${getColorClassName(columnColor as Color)}`}>
        <h2 className="note-dialog__header">{columnName}</h2>
        <div className={classNames("note-dialog__note", {"note-dialog__note--own-card": Parse.User.current()?.id === authorId})}>
          <div className="note-dialog__content">
            <blockquote className="note-dialog__text" contentEditable={editable} suppressContentEditableWarning onBlur={handleChangeNotetext}>
              {noteText}
            </blockquote>
          </div>

          <footer className="note-dialog__footer">
            <figure className="note-dialog__author">
              <img className="note-dialog__author-image" src={avatar} alt="User" />
              <figcaption className="note-dialog__author-name">{authorName}</figcaption>
            </figure>
          </footer>

          <aside>
            <ul className="note-dialog__options">
              <li className={classNames("note-dialog__option", {"note-dialog__option--not-editable": !editable})}>
                <IconButton onClick={handleDelete} direction="right" label="Delete" icon={deleteIcon} />
              </li>
            </ul>
          </aside>
        </div>
      </div>
    </Portal>
  );
};
export default NoteDialog;
