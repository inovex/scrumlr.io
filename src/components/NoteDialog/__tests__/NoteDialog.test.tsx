import {fireEvent, render, waitFor} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {wrapWithTestBackend} from "react-dnd-test-utils";
import {User} from "parse";
import {NoteDialog} from "components/NoteDialog";
import {mocked} from "ts-jest/utils";
import store from "store";
import {ActionFactory} from "store/action";

const mockStore = configureStore();
const mockedUser = mocked(User, true);

jest.mock("store", () => ({
  ...jest.requireActual("store"),
  dispatch: jest.fn(),
}));

type NoteDialogTestProps = {
  text?: string;
  authorId?: string;
  showAuthors?: boolean;
  activeModeration?: {userId: string; status: boolean};
  currentUserIsModerator?: boolean;
};

const createNoteDialog = (
  {text = "Test Text", authorId = "Test Author", showAuthors = true, activeModeration = {userId: authorId, status: false}, currentUserIsModerator = false}: NoteDialogTestProps = {
    text: "Test Text",
    authorId: "Test Author",
    showAuthors: true,
    activeModeration: {userId: "Test Author", status: false},
    currentUserIsModerator: false,
  }
) => {
  const initialState = {
    board: {
      data: {
        columns: [{id: "test_column", name: "test_header", hidden: false}],
      },
    },
    notes: [],
    users: {
      admins: [
        {
          id: "jkKqOUgt3hEDvl7CWcBokVOGs6AzINon",
          displayName: "Kinetic Kobold",
          admin: true,
          createdAt: "2021-08-11T10:45:41.640Z",
          updatedAt: "2021-08-11T10:52:21.558Z",
          online: true,
        },
      ],
      basic: [],
      all: [],
    },
  };
  const store = mockStore(initialState);
  const [NoteDialogContext] = wrapWithTestBackend(NoteDialog);
  return (
    <Provider store={store}>
      <NoteDialogContext
        noteId="0"
        text={text}
        authorId={authorId}
        columnName=""
        columnColor=""
        show
        activeVoting
        authorName=""
        showAuthors={showAuthors}
        votes={[
          {
            id: "test-id",
            board: "test-board",
            note: "0",
            user: "test-user",
            votingIteration: 1,
          },
        ]}
        childrenNotes={[
          {id: "1", columnId: "test_column", text: "", author: "", parentId: "0", dirty: true, authorName: "", votes: [], focus: false},
          {id: "2", columnId: "test_column", text: "", author: "", parentId: "0", dirty: true, authorName: "", votes: [], focus: false},
        ]}
        onClose={() => {}}
        activeModeration={activeModeration}
        currentUserIsModerator={currentUserIsModerator}
        onDeleteOfParent={() => {}}
      />
    </Provider>
  );
};

describe("<NoteDialog/>", () => {
  beforeEach(() => {
    window.IntersectionObserver = jest.fn(
      () =>
        ({
          observe: jest.fn(),
          disconnect: jest.fn(),
        } as unknown as IntersectionObserver)
    );

    const portal = global.document.createElement("div");
    portal.setAttribute("id", "portal");
    global.document.querySelector("body")!.appendChild(portal);
  });

  describe("should render correctly", () => {
    test("portal--darkBackground is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      expect(container.firstChild).toHaveClass("portal--darkBackground");
    });

    test("note-dialog is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      expect(container.querySelector(".portal__content")?.firstChild).toHaveClass("note-dialog");
    });

    test("note-dialog__note-header is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      expect(container.querySelector(".note-dialog")?.firstChild).toHaveClass("note-dialog__header");
    });

    test("three note-dialog__note present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
      expect(noteDialogNotes.length).toEqual(3);
    });

    test("note-dialog__note one is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
      expect(noteDialogNotes[0]).toHaveClass("note-dialog__note");
    });

    test("note-dialog__note two is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
      expect(noteDialogNotes[1]).toHaveClass("note-dialog__note");
    });

    test("note-dialog__note three is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
      expect(noteDialogNotes[2]).toHaveClass("note-dialog__note");
    });

    test("note-dialog__note-options note two is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
      expect(noteDialogNotes[1].querySelector(".note-dialog__options")).toBeDefined();
    });

    test("note-dialog__note-options note three is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
      expect(noteDialogNotes[2].querySelector(".note-dialog__options")).toBeDefined();
    });

    describe("NoteDialogNoteContent", () => {
      test("note-dialog__note-content is present", () => {
        const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
        expect(container.querySelector(".note-dialog__note")!.querySelector(".note-dialog__note-content")).toBeDefined();
      });

      test("click on NoteContent as author", async () => {
        mockedUser.current = jest.fn(() => ({id: "test-user-2"} as never));
        const {container} = render(createNoteDialog({authorId: "test-user-2"}), {container: global.document.querySelector("#portal")!});
        const noteContentText = container.querySelector(".note-dialog__note-content__text")!;
        fireEvent.blur(noteContentText, {target: {textContent: "Changed Text"}});

        await waitFor(() => {
          expect(store.dispatch).toHaveBeenCalledWith(ActionFactory.editNote({id: "0", text: "Changed Text"}));
        });
      });

      test("click on NoteContent as moderator", async () => {
        mockedUser.current = jest.fn(() => ({id: "test-user-2"} as never));
        const {container} = render(createNoteDialog({currentUserIsModerator: true}), {container: global.document.querySelector("#portal")!});
        const noteContentText = container.querySelector(".note-dialog__note-content__text")!;
        fireEvent.blur(noteContentText, {target: {textContent: "Changed Text"}});

        await waitFor(() => {
          expect(store.dispatch).toHaveBeenCalledWith(ActionFactory.editNote({id: "0", text: "Changed Text"}));
        });
      });

      test("click on NoteContent as other user", async () => {
        mockedUser.current = jest.fn(() => ({id: "test-user-2"} as never));
        const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
        const noteContentText = container.querySelector(".note-dialog__note-content__text")!;
        fireEvent.blur(noteContentText, {target: {textContent: "Changed Text"}});

        await waitFor(() => {
          expect(store.dispatch).not.toHaveBeenCalled();
        });
      });
    });

    test("note-dialog__note-footer is present", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      expect(container.querySelector(".note-dialog__note")!.querySelector(".note-dialog__note-footer")).toBeDefined();
    });

    test("note-dialog match snapshot", () => {
      const {container} = render(createNoteDialog(), {container: global.document.querySelector("#portal")!});
      expect(container.querySelector(".note-dialog")).toMatchSnapshot();
    });

    describe("Moderation phase", () => {
      beforeEach(() => {
        mockedUser.current = jest.fn(() => ({id: "test-user-2"} as never));
      });

      test("three note-dialog__note present during moderation phase", () => {
        const {container} = render(createNoteDialog({activeModeration: {userId: "test-user-1", status: true}, authorId: "test-user-1"}), {
          container: global.document.querySelector("#portal")!,
        });
        const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
        expect(noteDialogNotes.length).toEqual(3);
      });

      test("moderation: last note-dialog__options note isn't present", () => {
        const {container} = render(createNoteDialog({activeModeration: {userId: "test-user-1", status: true}, authorId: "test-user-1"}), {
          container: global.document.querySelector("#portal")!,
        });
        const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
        expect(noteDialogNotes[noteDialogNotes.length - 1].children.length).toEqual(2);
        expect(noteDialogNotes[noteDialogNotes.length - 1].querySelector(".note-dialog__note-content")).not.toBeNull();
        expect(noteDialogNotes[noteDialogNotes.length - 1].querySelector(".note-dialog__note-footer")).not.toBeNull();
        expect(noteDialogNotes[noteDialogNotes.length - 1].querySelector(".note-dialog__note-options")).toBeNull();
      });

      test("moderation: last note-dialog__notes has two children", () => {
        const {container} = render(createNoteDialog({activeModeration: {userId: "test-user-1", status: true}, authorId: "test-user-1"}), {
          container: global.document.querySelector("#portal")!,
        });
        const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
        expect(noteDialogNotes[noteDialogNotes.length - 1].children.length).toEqual(2);
      });

      test("moderation: last note-dialog__options note is present", () => {
        const {container} = render(createNoteDialog({activeModeration: {userId: "test-user-1", status: true}, authorId: "test-user-1", currentUserIsModerator: true}), {
          container: global.document.querySelector("#portal")!,
        });
        const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
        expect(noteDialogNotes[noteDialogNotes.length - 1].querySelector(".note-dialog__note-content")).not.toBeNull();
        expect(noteDialogNotes[noteDialogNotes.length - 1].querySelector(".note-dialog__note-footer")).not.toBeNull();
        expect(noteDialogNotes[noteDialogNotes.length - 1].querySelector(".note-dialog__note-options")).not.toBeNull();
      });

      test("moderation: last note-dialog__notes has three children", () => {
        const {container} = render(createNoteDialog({activeModeration: {userId: "test-user-1", status: true}, authorId: "test-user-1", currentUserIsModerator: true}), {
          container: global.document.querySelector("#portal")!,
        });
        const noteDialogNotes = container.querySelectorAll(".note-dialog__note");
        expect(noteDialogNotes[noteDialogNotes.length - 1].children.length).toEqual(2);
      });
    });
  });
});
