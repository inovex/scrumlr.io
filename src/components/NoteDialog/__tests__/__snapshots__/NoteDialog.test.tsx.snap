// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteDialog/> should render correctly note-dialog match snapshot 1`] = `
<div
  class="note-dialog accent-color__"
>
  <h2
    class="note-dialog__header"
  />
  <div
    class="note-dialog__scrollbar"
  >
    <div
      class="note-dialog__note-wrapper"
    >
      <div
        class="note-dialog__note"
      >
        <div
          class="note-dialog__note-content"
        >
          <blockquote
            class="note-dialog__note-content__text"
            contenteditable="false"
          >
            Test Text
          </blockquote>
        </div>
        <div
          class="note-dialog__note-footer"
        >
          <figure
            class="note-dialog__note-author"
          >
            <div
              class="user-avatar note-dialog__note-user-avatar"
            >
              <div
                class="user__initials"
                title=""
              />
            </div>
            <figcaption
              class="note-dialog__note-author-name"
            />
          </figure>
          <div
            class="note-dialog__note-footer__options-and-votes"
          >
            <div
              class="votes"
            >
              <button
                class="dot-button vote-button-remove"
              >
                <span
                  class="vote-button-remove__folded-corner"
                />
                <span>
                  1
                </span>
              </button>
              <button
                class="dot-button vote-button-add"
              >
                <svg
                  class="vote-button-add__icon"
                >
                  icon-add.svg
                </svg>
              </button>
            </div>
            <ul
              class="note-dialog__note-options"
            />
          </div>
        </div>
      </div>
      <div
        class="note-dialog__note"
      >
        <div
          class="note-dialog__note-content"
        >
          <blockquote
            class="note-dialog__note-content__text"
            contenteditable="false"
          />
        </div>
        <div
          class="note-dialog__note-footer"
        >
          <figure
            class="note-dialog__note-author"
          >
            <div
              class="user-avatar note-dialog__note-user-avatar"
            >
              <div
                class="user__initials"
                title=""
              />
            </div>
            <figcaption
              class="note-dialog__note-author-name"
            />
          </figure>
          <div
            class="note-dialog__note-footer__options-and-votes"
          >
            <div
              class="votes"
            >
              <button
                class="dot-button vote-button-add"
              >
                <svg
                  class="vote-button-add__icon"
                >
                  icon-add.svg
                </svg>
              </button>
            </div>
            <ul
              class="note-dialog__note-options"
            >
              <li
                class="note-dialog__note-option"
              >
                <button
                  class="dot-button note-dialog__note-option__unstack"
                >
                  <svg
                    class="note-dialog__note-option__unstack-icon"
                  >
                    icon-unstack.svg
                  </svg>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="note-dialog__note"
      >
        <div
          class="note-dialog__note-content"
        >
          <blockquote
            class="note-dialog__note-content__text"
            contenteditable="false"
          />
        </div>
        <div
          class="note-dialog__note-footer"
        >
          <figure
            class="note-dialog__note-author"
          >
            <div
              class="user-avatar note-dialog__note-user-avatar"
            >
              <div
                class="user__initials"
                title=""
              />
            </div>
            <figcaption
              class="note-dialog__note-author-name"
            />
          </figure>
          <div
            class="note-dialog__note-footer__options-and-votes"
          >
            <div
              class="votes"
            >
              <button
                class="dot-button vote-button-add"
              >
                <svg
                  class="vote-button-add__icon"
                >
                  icon-add.svg
                </svg>
              </button>
            </div>
            <ul
              class="note-dialog__note-options"
            >
              <li
                class="note-dialog__note-option"
              >
                <button
                  class="dot-button note-dialog__note-option__unstack"
                >
                  <svg
                    class="note-dialog__note-option__unstack-icon"
                  >
                    icon-unstack.svg
                  </svg>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
